drawBot/drawBotDrawingTools.py

```diff
@@ -180,6 +199,27 @@ def endDrawing(self):
         self._uninstallAllFonts()
         gifTools.clearExplodedGifCache()
 
+    def drawing(self):
+        """
+        Reset and clean the drawing stack in a `with` statement.
+
+        .. downloadcode:: drawing.py
+
+            # Use the 'with' statement.
+            # This will make sure that the stack of pages is cleaned and reset
+            # once the interpreter exits the `with` statement
+            # The following example shows how to create three PDF booklets
+            # and it uses the `with drawing()` statement to ensure that page numbers
+            # restart from 1 in each PDF
+            for eachBooklet in range(1, 4):
+                with drawing():
+                    for eachPage in range(10):
+                        newPage(400, 400)
+                        text(f"{pageCount()}", (100, 100))
+                    saveImage(f"book_{eachBooklet}.pdf")
+        """
+        return DrawingContextManager(self)
```

Comment:
Suggest to use a contextlib.contextmanager decorator on this function instead:

- https://docs.python.org/3/library/contextlib.html#contextlib.contextmanager

Also, we'll need an "expected result" image for the demo code. I forgot what the easiest way to do that is.